src/github.com/couchbase/sync_gateway/db/change_cache.go

```diff
@@ -306,6 +306,27 @@ func (c *changeCache) DocChanged(docID string, docJSON []byte) {
 			c.processEntry(change)
 		}
 
+		// If the recent sequence history includes any sequences earlier than the current sequence, and
+		// not already seen by the gateway (more recent than c.nextSequence), add them as empty entries.
+		currentSequence := doc.Sequence
+		if len(doc.UnusedSequences) > 0 {
```

Comment:
Can you add some docs on the doc.UnusedSequences field to describe what these are exactly?  The name is descriptive, but not complete in the sense that I still wonder: who is using them, and why are they unused?